File: GraffiTab_iOS_SDK/Classes/Model/GTComment.m
//
//  Comment.m
//  GraffiTab-iOS
//
//  Created by Georgi Christov on 04/12/2014.
//  Copyright (c) 2014 GraffiTab. All rights reserved.
//

#import "GTComment.h"

@implementation GTComment

- (id)initFromJson:(NSDictionary *)json {
    self = [super init];
    
    if (self) {
        self.commentId = [json[JSON_COMMENT_ID] longValue];
        self.itemId = [json[JSON_COMMENT_ITEMID] longValue];
        self.user = [[GTPerson alloc] initFromJson:json[JSON_COMMENT_USER]];
        self.text = json[JSON_COMMENT_TEXT];
        self.date = [NSDate dateWithTimeIntervalSince1970:[json[JSON_COMMENT_DATE] longValue] / 1000];
    }
    
    return self;
}

- (NSDictionary *)asDictionary {
    NSMutableDictionary *json = [NSMutableDictionary new];
    
    json[JSON_COMMENT_ID] = @(self.commentId);
    json[JSON_COMMENT_ITEMID] = @(self.itemId);
    json[JSON_COMMENT_USER] = self.user.asDictionary;
    json[JSON_COMMENT_TEXT] = self.text;
    json[JSON_COMMENT_DATE] = @(self.date.timeIntervalSince1970);
    
    return json;
}

- (BOOL)isEqual:(id)other {
    if (other == self)
        return YES;
    if (!other || ![other isKindOfClass:[self class]])
        return NO;
    return [self isEqualToWidget:other];
}

- (BOOL)isEqualToWidget:(GTComment *)aWidget {
    if (self == aWidget)
        return YES;
    if (self.commentId != aWidget.commentId)
        return NO;
    return YES;
}

@end

File: GraffiTab-iOS-SDK/Model/GTComment.swift
//
//  GTComment.swift
//  GraffiTab-iOS-SDK
//
//  Created by Georgi Christov on 11/04/2016.
//  Copyright © 2016 GraffiTab. All rights reserved.
//

import UIKit
import ObjectMapper

public class GTComment: NSObject, Mappable {

    public var id: Int?
    public var streamable: GTStreamable?
    public var user: GTUser?
    public var text: String?
    public var createdOn: NSDate?
    public var updatedOn: NSDate?
    
    required public init?(_ map: Map) {
        
    }
    
    public func mapping(map: Map) {
        id <- map["id"]
        streamable <- map["streamable"]
        user <- map["user"]
        text <- map["text"]
        createdOn <- (map["createdOn"], CustomDateFormatTransform(formatString: GTApiDateConstants.InputFormat))
        updatedOn <- (map["updatedOn"], CustomDateFormatTransform(formatString: GTApiDateConstants.InputFormat))
    }
    
    public func softCopy(other: GTComment) {
        self.id = other.id
        self.streamable?.softCopy(other.streamable!)
        self.user?.softCopy(other.user!)
        self.text = other.text
        self.createdOn = other.createdOn
        self.updatedOn = other.updatedOn
    }
    
    override public func isEqual(object: AnyObject?) -> Bool {
        if let object = object as? GTComment {
            return id == object.id
        } else {
            return false
        }
    }
    
    override public var hash: Int {
        return id!.hashValue
    }
}
